src/visualization/makie_visualization.jl

```diff
@@ -178,76 +184,91 @@ function generate_plotting_triangulation(pds::PlotDataSeries2D{<:UnstructuredPlo
   return plotting_mesh
 end
 
-function generate_plotting_wireframe(pds::PlotDataSeries2D{<:UnstructuredPlotData2D};
+# Returns a list of `Makie.Point`s which can be used to plot the mesh, or a solution "wireframe"
+# (e.g., a plot of the mesh lines but with the z-coordinate equal to the value of the solution).
+function mesh_plotting_wireframe(pds::PlotDataSeries2D{<:UnstructuredPlotData2D};
                                      set_z_coordinate_zero = false)
   @unpack variable_id = pds
   pd = pds.plot_data
   @unpack xf, yf, uf = pd
 
-  if set_z_coordinate_zero==true
-    sol_f = zeros(size(xf)) # plot 2d surface by setting z coordinate to zero
+  if set_z_coordinate_zero
+    sol_f = zeros(eltype(first(uf)), size(xf)) # plot 2d surface by setting z coordinate to zero
   else
     sol_f = StructArrays.component(uf, variable_id)
   end
   xyz_wireframe = Makie.Point.(map(x->vec(vcat(x, fill(NaN, 1, size(x, 2)))), (xf, yf, sol_f))...)
+
   return xyz_wireframe
 end
 
+function default_nvisnodes(sol::TrixiODESolution)
+  semi = sol.prob.p
+  dg = semi.solver
+  return 2*length(dg.basis.nodes)
+end
+
 """
-    trixi_plot(sol::TrixiODESolution;
+    iplot(sol::TrixiODESolution;
                solution_variables=nothing, nvisnodes=5, variable_to_plot_in = 1)
 
-Creates an interactive surface plot of the solution and mesh.
+Creates an interactive surface plot of the solution and mesh for an `UnstructuredMesh2D` type.
 
 Inputs:
 - solution_variables: either `nothing` or a variable transformation function (e.g., `cons2prim`)
 - nvisnodes: number of visualization nodes per dimension
 """
-function trixi_plot(sol::TrixiODESolution;
-                    solution_variables=nothing, nvisnodes=5, variable_to_plot_in = 1)
+function iplot(sol::TrixiODESolution;
+               solution_variables=nothing, nvisnodes=default_nvisnodes(sol), variable_to_plot_in = 1)
```

Comment:
What about using the signature `iplot(sol)` just as convenience version redirecting the work to `iplot(u, semi)` (or whatever signature you need)? This would make it much more convenient when working interactively, since we can copy and modify `u` instead of needing to create an ODE solution and modify it in place. Feel free to create an issue for this instead of implementing it here right now.

Reply:
I second this. That's what we also did for Plots.jl - the `sol` stuff is only syntactic sugar, redirecting to `u_ode, semi`, redirecting to the main implementation, `u, semi`.

Reply:
Isn't the last one `u, mesh, equations, solver, cache` or something like this?

Reply:
You are - as usual - correct: https://github.com/trixi-framework/Trixi.jl/blob/9364f7fe5c5379c76e3a1fbeb5fcdd99dcd078cc/src/visualization/plot_recipes.jl#L111-L114

Thus, the sugar goes `sol` -> `u_ode, semi` -> `u, mesh, equations, solver, cache`. I think it would be nice if this were analogous for Makie.jl plotting.